apps/postgres.go

```diff
@@ -0,0 +1,274 @@
+func (pb *postgresBuilder) WithPvcSize(sizeGi int) *postgresBuilder {
+	pb.pvcSize = sizeGi
+	pb.values["persistence.size"] = fmt.Sprintf("%dGi", sizeGi)
+	return pb
+}
+
+// WithReleaseName Could specify helm release name - default = "maya-mongo"
+func (pb *postgresBuilder) WithReleaseName(releaseName string) *postgresBuilder {
+	pb.releaseName = releaseName
+	return pb
+}
+
+// WithHelmChartVersion Could specify helm chart version - default = "latest"
+func (pb *postgresBuilder) WithHelmChartVersion(chartVersion string) *postgresBuilder {
+	pb.helmVersion = chartVersion
+	return pb
+}
+
+// WithFileSystemType Could specify volume provisioning type - default = "thin"
+func (pb *postgresBuilder) WithFileSystemType(systemType common.FileSystemType) *postgresBuilder {
+	pb.filesystemType = systemType
+	return pb
+}
+
+// WithOwnStorageClass Could specify already created storage class name.
+// If this option is selected, you must ensure that the storage class parameters are set correctly.
+func (pb *postgresBuilder) WithOwnStorageClass(scName string) *postgresBuilder {
+	pb.scName = scName
+	pb.values["global.storageClass"] = scName
+	return pb
+}
+
+// WithPvc Could specify pvc
+// If this option is selected, you must ensure that the pvc is created correctly.
+func (pb *postgresBuilder) WithPvc(pvcName string) *postgresBuilder {
+	if arch, ok := pb.values["architecture"].(string); ok {
+		if arch == Standalone.String() {
+			pb.pvcName = pvcName
+			pb.values["primary.persistence.existingClaim"] = pvcName
+			return pb
+		}
+	}
+	return pb
+}
+
+// WithAnotherValuesParameters Could specify another parameters from https://artifacthub.io/packages/helm/bitnami/postgresql
+func (pb *postgresBuilder) WithAnotherValuesParameters(values map[string]interface{}) *postgresBuilder {
+	for k, v := range values {
+		pb.values[k] = v
+	}
+	return pb
+}
+
+// WithPgBench Also install pg_bench benchmark app
+func (pb *postgresBuilder) WithPgBench() *postgresBuilder {
+	pb.pgBench = true
+	return pb
+}
+
+func (pb *postgresBuilder) Build() (PostgresApp, error) {
+	var latest Chart
+	latest, err := GetLatestHelmChartVersion(e2e_config.GetConfig().Product.PostgresHelmRepo)
+	if err != nil && pb.helmVersion == "" {
+		logf.Log.Error(err, "switching to default bitnami/postgresql chart version", "defaultChart", e2e_config.GetConfig().Product.PostgresDefaultChartVersion)
+		pb.helmVersion = e2e_config.GetConfig().Product.PostgresDefaultChartVersion
+	}
+	if pb.helmVersion == "" {
+		pb.helmVersion = latest.Version
+	}
+	if pb.scName == "" {
+		scName, err := CreatePostgresStorageClass(pb)
+		if err != nil {
+			return PostgresApp{}, err
+		}
+		pb.values["global.storageClass"] = scName
+		pb.scName = scName
+		logf.Log.Info("StorageClass has been created", "storageClassName", scName)
+	}
+	err = AddHelmRepository(e2e_config.GetConfig().Product.PostgresHelmRepoName, e2e_config.GetConfig().Product.PostgresHelmRepoUrl)
```

Comment:
Build and Install should be separate

Reply:
In my opinion, a build in this case means that you have created a ready-made postgres application and I'll stick with keeping them together, but we can discuss this because a build without installation is unnecessary in this case.

Reply:
Done